dataikuapi/dss/ml.py

```diff
@@ -946,7 +984,7 @@ def _repr_html_(self):
         return res + "<details><pre>{}</pre></details>".format(self.__repr__())
 
     def __repr__(self):
-        return self.__class__.__name__ + "(values={})".format(super(PredictionAlgorithmSettings, self).copy())
+        return self.__class__.__name__ + "(values={})".format(self.copy())
```

Comment:
Why a copy ?

Reply:
My guess: because `copy` produces a dict with its regular string repr, while `self` would look to convert `self` to a string, recursively calling `__repr__`, causing infinite recursion.

Reply:
Then maybe:
```suggestion
        return self.__class__.__name__ + "(values={})".format(dict.__repr__(self))
```
or
```suggestion
        return self.__class__.__name__ + "(values={})".format(super(PredictionAlgorithmSettings, self).__repr__())
```

Reply:
> My guess: because `copy` produces a dict with its regular string repr, while `self` would look to convert `self` to a string, recursively calling `__repr__`, causing infinite recursion.

Also: when calling `"{}".format(something)` this calls `something.__str__()`. In this case it is the same as `something.__repr__()` because `__str__ = __repr__` when it is not explicitly implemented